### DISCUSSION TITLE
Automatically create s3 backend?

### DISCUSSION THREAD

**Discussion by jmreicha** [gruntwork-io/knowledge-base#472] [Help] [answered]
I am running into an issue when attempting to start a new environment from scratch, which requires a new remote state backend to be created.

The problem only appears when I do a `run-all` from a directory that has dependencies. I have tested and get prompted to create the remote state if I do a `run-all` from a directory that has no dependencies and get prompted with `Remote state S3 bucket foo does not exist or you don't have permissions to access it. Would you like Terragrunt to create it? (y/n)`

From what I can see there is some discrepancy in how dependencies are treated, but my goal would be to be able to do a `run-all` and spin up everything in one shot with no manual intervention. Any guidance here?

---

<ins datetime="2022-06-16T14:01:47Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108788">Tracked in ticket #108788</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
If you are starting in a new environment where the corresponding S3 Bucket for the state is not yet available, and you wish to always automatically create it, you can call `terragrunt` with the `--terragrunt-non-interactive` flag to always precreate the buckets. Note that if you are concerned about `--terragrunt-non-interactive` saying `yes` to a prompt without your input, you can run `terragrunt run-all init --terragrunt-non-interactive` for the bucket setup, which is a safe operation.

**Reply by jmreicha** [reply]
Ah yep. I also noticed that setting `skip_outputs` in the dependency block also needs to be set.

Thank you for the quick response!
